lib/logstash/outputs/mongodb.rb

```diff
@@ -38,20 +41,18 @@ def register
     @db = conn.use(@database)
   end # def register
 
-  public
   def receive(event)
-    
-
     begin
-      if @isodate
-        # the mongodb driver wants time values as a ruby Time object.
-        # set the @timestamp value of the document to a ruby Time object, then.
-        document = event.to_hash
-      else
-        document = event.to_hash.merge("@timestamp" => event["@timestamp"].to_json)
+      # Our timestamp object now has a to_bson method, using it here
+      # why not use to_hash_with_metadata?
+      # {}.merge(other) so we don't taint the event hash innards
+      document = {}.merge(event.to_hash)
```

Comment:
My thinking is this:
- `event.to_hash` is a non standard method on an object that this class knows nothing about.
- there is no compiler/interface certainty that `event.to_hash` will return a Ruby Hash, it only has to be hash-like by convention.
- there is no certainty that the object returned by `to_hash` is clonable.
- there is no certainty about how the `clone` is implemented.
- By using `{}.merge`, I am establishing three things: 1) document will be a Ruby Hash, 2) it is independent from the event, 3) the object given to merge only needs to implement `each {|k,v| ...}`.

All in all, it makes this code less dependent on the Event implementation, now and in the future.